src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,14 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * Checks if command mutates data in the address book.
+     *
+     * @return true is command mutates data
+     */
+    public boolean isMutating() {
+        //check is done in subclasses
+        throw new UnsupportedOperationException("This method should be implement in child classes");
```

Comment:
Throwing exceptions is one way; but doesn't serve the purpose as someone can circumvent implementing the method altogether by catching the exception and not doing anything. The nice way to force implementing the method is to make it abstract.